server/protocols/jmap/src/main/java/org/apache/james/jmap/mailet/filter/HeaderExtractor.java

```diff
@@ -75,7 +76,8 @@ static HeaderExtractor addressExtractor(ThrowingFunction<Mail, Address[]> addres
             } catch (Exception e) {
                 LOGGER.info("Failed parsing header. Falling back to unparsed header value matching", e);
                 return Stream.of(mail.getMessage().getHeader(fallbackHeaderName))
-                    .map(MimeUtil::unscrambleHeaderValue);
+                    .map(MimeUtil::unscrambleHeaderValue)
+                    .flatMap(s -> Splitter.on(',').trimResults().splitToList(s).stream());
```

Comment:
I don't think you can split on "," for mail addresses. For example `"a, b" <ab@example.com>, c@example.com` is a valid address list but will fail in this case.

Reply:
Well, the scenario that you present is very rare.

We are in "recovery mode" and handling multi-valued mails seems more important to me than handling weird syntax. Note that when javax.mail does not fail we rely on its parsing and thus handle such cases.

So here the question IMO is: *When javax.mail fails do we prefer*:
 - Handling very rare syntax?
 - Handling very common case of several recipients specified in the same header?

Personally I prefer the second.

Reply:
there are a lot of options in javax.mail about parsing. See for example https://docs.oracle.com/javaee/6/api/javax/mail/internet/package-summary.html

Can you check if one of them allows to bypass this error?

Reply:
Here you are.

Please note that I did not want system-wide parameters for what is a local concern. I prefer relying on some parsing utilities from javax for this.